superset-frontend/cypress-base/cypress/integration/explore/control.test.ts

```diff
@@ -37,7 +37,7 @@ describe('Datasource control', () => {
     cy.get('#datasource_menu').click();
     cy.get('a').contains('Edit Datasource').click();
     // create new metric
-    cy.get('button').contains('Add Item').click();
+    cy.get('table button').contains('Add Item', { timeout: 10000 }).click();
```

Comment:
FWIW, I'm planning to go through and replace lots of these cypress DOM selectors with `data-test` attribute selectors, so things are more explicit and easier to maintain test integrity/auditability. If you have feelings on that, let me know :)

Reply:
TBH I'm not a big fan of the `data-test` attributes since it exposes dev stuff to production---not that it matters much though. I personally prefer HTML tag selector + `contains(...)` because: 1) they are more or less more stable than CSS class names, 2) tests probably do need to break when copy text or underlying data changes.

Reply:
Fair point about production noise... I was thinking we should follow up and clean them out using something like [this](https://www.npmjs.com/package/babel-plugin-jsx-remove-data-test-id). But I don't want to hijack your PR threads for this discussion ;) You just made it cross my mind!